lib/gems/pending/util/miq_ftp_lib.rb

```diff
@@ -56,7 +56,10 @@ def create_directory_structure(directory_path)
       end
       ftp.chdir(directory)
     end
-    ftp.chdir(pwd)
+  rescue Net::FTPPermError => e
+    _log.info("introspection of directories disabled. skipping create_directory_structure")
+  ensure
+    ftp.chdir(pwd) if pwd
```

Comment:
Can we remove the `if pwd` here?  The `pwd` should always be set, so want to always make sure we are returning to that directory.

Reply:
I feel more comfortable having the `if pwd`

It shouldn't be nil. but there is a chance that it gets nil if the first line throws an exception.

with the `if` then this line will be ignored, but the original exception will continue through.
if we remove the `if` here, then this will throw an unrelated (and meaningless(?) exception) - hiding the original exception

Reply:
I am fine with that explanation.  Not standard practice I would guess, but makes sense to me why you are doing this now, so keep it as is.  👍